### PR TITLE
Improve Tests OCBytecodeToASTCache

### PR DESCRIPTION
**Pull request by MarcusDenker**
OCBytecodeToASTCacheTest had some tests that tested against the old implementation. As we replaced this with the cache, we need to rewite the tests to compare something else (e.g which kind of node)

### CODE REVIEW

File: src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st [hunk 1]
```diff
@@ -60,21 +61,17 @@ OCBytecodeToASTCacheTest >> testLowerThanFirstBCOffsetAccessTest [
...
-	we have to check that associations are consistent between a range and a node"
+	we have to check that associations are all pointing to a node"
 
 	| pcRange |
 	pcRange := 0 to: cache lastBcOffset.
-	pcRange do: [ :pc | 
-		self
-			assert: (cache nodeForPC: pc)
-			identicalTo: (compiledMethod sourceNodeForPC: pc) ]
+	pcRange do: [ :pc | self assert: ((cache nodeForPC: pc) isKindOf: RBProgramNode)]
```
Comment (StevenCostiou) [inline src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st:68]: I agree in the end that is better, however we are only testing that it maps to a node but not the correct one.
I would do a brute check: for each pc, check that it is the correct node.
Maybe we can generate it?

File: src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st [hunk 2]
```diff
@@ -60,21 +61,17 @@ OCBytecodeToASTCacheTest >> testLowerThanFirstBCOffsetAccessTest [
...
 
 { #category : #tests }
 OCBytecodeToASTCacheTest >> testNodeForBCOffsetTest [
 	| pc |
+	"there should be an assignment at pc 51"
 	pc := 51.
-	self
-		assert: (cache nodeForPC: pc)
-		identicalTo: (compiledMethod sourceNodeForPC: pc)
+	self assert: (cache nodeForPC: pc) isAssignment
```
Comment (StevenCostiou) [inline src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st]: I would add additional checks. I put that example here thinking that it refers to the assignment `i := 5` of `helper12` method (but same idea if that's the other assignment).

```Smalltalk
| pc node |
"there should be an assignment at pc 51"	
pc := 51.
node := (cache nodeForPC: pc).
self assert: node isAssignment.
self assert: node variable equals: i. "find the i var node in the method"
self assert: node variable equals: 5. "put a literal node here to check for equality"
```

Comment (MarcusDenker) [reply src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st]: I change it to: 	

self assert: (cache nodeForPC: pc) sourceCode equals: 'i := i - 1'.

File: src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st [hunk 3]
```diff
@@ -73,5 +73,5 @@ OCBytecodeToASTCacheTest >> testNodeForBCOffsetTest [
 	| pc |
 	"there should be an assignment at pc 51"
 	pc := 51.
-	self assert: (cache nodeForPC: pc) isAssignment
+	self assert: (cache nodeForPC: pc) sourceCode equals: 'i := i - 1'.
```
Comment (StevenCostiou) [inline src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st:76]: I would also check:
- node class
- node start
- node stop
This way we're sure it is the correct one, as source code strings can be printed from different nodes.

File: src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st (hunk 1 above)
Comment (StevenCostiou) [reply src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st:68]: I generated a test for this method: https://github.com/StevenCostiou/OpalASTMappingTestGenerator

Comment (MarcusDenker) [reply src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st:68]: I add this to the backlog, I would like to merge the current state

File: src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st (hunk 3 above)
Comment (MarcusDenker) [reply src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st:76]: There is just one expression "i := i - 1" in th method under test. there is no way that we print that string and it is not the node we are looking for

Comment (StevenCostiou) [reply src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st:76]: But I'm concerned by the fact that we may change the test method one day, and introduce such subtility that would make the test pass while it should not.
As we're testing the node mapping, I think we need to always make sure that we precisely identify the node we test.

Comment (MarcusDenker) [reply src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st:76]: ok, added stop and start check

**Comment by MarcusDenker** [comment]
I integrate this, next steps will be another PR

**Pull request merged**